### PR TITLE
Testcases

### PR DESCRIPTION
**Pull request by Sowmayjain**
Our smart contracts are finally completed and ready for audits. We had started talking to audit firms so we will send out the contracts in coming days.

@leckylao could you write the test-cases for all the contracts? You can also check the previous [flusher repo](https://github.com/InstaDApp/yield-flusher) where you wrote some test-cases, previously.

### CODE REVIEW

File: contracts/tests/daiLogic.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    
+    TokenInterface baseToken;
+
+    function getTotalToken() public returns (uint) {
+        uint bal = baseToken.balanceOf(address(this));
+        bal += baseToken.balanceOf(address(poolToken));
+        return bal;
+    }
+
+    function reduceDai(uint amt) public {
```
Comment (leckylao) [inline contracts/tests/daiLogic.sol:20]: Anyone can call or only owner?

File: contracts/tests/ethLogic.sol [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+contract EthRateLogic {
+    address poolToken;
+    
+    function getTotalToken() public returns (uint) {
+        uint bal = (address(this).balance);
+        bal += (address(poolToken).balance);
+        return bal;
+    }
+
+    function reduceETH(uint amt) public {
```
Comment (leckylao) [inline contracts/tests/ethLogic.sol:13]: Anyone can call or only owner?

File: contracts/tests/daiLogic.sol (hunk 1 above)
Comment (thrilok209) [reply contracts/tests/daiLogic.sol:20]: These are just test rate logic contracts. To test out different conditon. Reduce and increase token supply to see different behaviour of the contracts

**Comment by thrilok209** [comment]
image

**Pull request merged**